### PR TITLE
feature/474

### LINKED ISSUE #474: Airport Revival :: Flughafen Wien-Schwechat
Vienna International Airport (German: Flughafen Wien-Schwechat; IATA: VIE, ICAO: LOWW) is the international airport of Vienna, the capital of Austria, located in Schwechat, 18 km (11 mi) southeast of central Vienna and 57 km west of Bratislava. It is the country's largest airport and serves as the hub for Austrian Airlines and a base for Eurowings Europe. It is capable of handling wide-body aircraft up to the Airbus A380. The airport features a dense network of European destinations as well as long-haul flights to Asia, North America and Africa. In 2017, the airport handled 24.4 million passengers, a 4.5% increase compared to 2016.

- [x] basic airport information
- [ ] new terrain
- [ ] new airspace
- [ ] new fixes
- [ ] restricted areas
- [ ] airways
- [ ] SIDs
- [ ] STARs
- [ ] spawnPatterns
- [ ] videomap

### PR DESCRIPTION
**Pull request by jakobeng1303**
Resolves #474.

1. News SIDS
2. New STARS/Transtions
3. New Wind -> new config: 29dep, 34arr

Because I was never able to get my own copy of this game running, I hope someone can test this

### CODE REVIEW

**Comment by n8rzz** [comment]
I'm spinning up a review app now so you can test it.

**Comment by jakobeng1303** [comment]
It seems like I'm not having access to the heroku file (access denied)

File: assets/airports/loww.json [hunk 1]
```diff
@@ -458,15 +455,15 @@
...
                 "11": [["WW151", "S205-"], "WW152", "WW170"],
-                "29": [["WW293", "A10|S205-"], ["WW230", "A40+"], "WW266"],
+                "29": [["WW293", "A10+|S205-"], ["WW230", "A40+"], "WW266"],
                 "16": [["WW267", "S205-"], "WW160", "WW170"],
-                "34": [["WW165", "A15|S205-"], "WW371", "WW374"]
+                "34": [["WW165", "A15+|S205-"], "WW371", "WW374"]
             },
             "exitPoints": {
-                "LANUX": ["WW171", "LANUX"]
+                "LANUX": ["WW171", ["LANUX", "A240-"]
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing a bracket @jakobeng1303

Comment (indianbhaji) [inline assets/airports/loww.json]: This should be `"LANUX": ["WW171", ["LANUX", "A240-"]]`

Comment (jakobeng1303) [inline assets/airports/loww.json]: thanks!

File: assets/airports/loww.json [hunk 2]
```diff
@@ -590,244 +572,94 @@
...
-            },
-            "body": [["REKLU", "A180|S250-"], ["MABOD", "A60-|S250"]],
-            "draw": [["REKLU", "MABOD"]]
+        "11": {
+          "icao": "11",
+          "name": "runway 11",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A80|S250-"], ["WW666", "A80-|S250-"], ["WW664", "A50|S250"], ["WW658", "A50|S250-"], ["WW656", "A50-|S250-"], ["WW654", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW664", "A50|S250"], ["WW658", "A50|S250-"], ["WW656", "A50-|S250-"], ["WW654", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW669", "A60+|S250-"], ["WW665", "A60|S250-"], ["WW663", "A50|S250"], ["WW661", "A50|S250-"], ["WW659", "A50-|S250-"], ["WW657", "A50-|S250-"], ["WW655", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Colon missing, should be `BALAD :`

File: assets/airports/loww.json [hunk 3]
```diff
@@ -590,244 +572,94 @@
...
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW667", "A80|S250-"], ["WW665", "A60|S250-"], ["WW663", "A50|S250"], ["WW661", "A50|S250-"], ["WW659", "A50-|S250-"], ["WW657", "A50-|S250-"], ["WW655", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Colon missing, should be `PESAT:`

File: assets/airports/loww.json [hunk 4]
```diff
@@ -590,244 +572,94 @@
...
+              "MASUR": ["_MASUR272009", "MASUR", "A170-|S330"], ["TEMTA", "A60-|S250"], ["NERDU", "A80|S250-"], ["WW666", "A80-|S250-"], ["WW664", "A50|S250"], ["WW658", "A50|S250-"], ["WW656", "A50-|S250-"], ["WW654", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing bracket, should be `["MASUR, "A170-|S330"]`

File: assets/airports/loww.json [hunk 5]
```diff
@@ -590,244 +572,94 @@
...
+            },
+          "body": [["WW653", "A50-|S250-"], ["WW652", "A50-|S250-"], ["WW651", "A50-|S250-"], ["WW650", "A50-|S250-"], ["RW11", "A50-|S250-"]]
+        },
+        "29": {
+          "icao": "29",
+          "name": "runway 29",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A60+|S250-"], ["WW963", "A70+|S250"], ["WW961", "A70|S250-"], ["WW959", "A60|S250-"], ["WW957", "A60-|S250-"], ["WW955", "A60-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW967", "A60+|S250-"], ["WW965", "A60+|S250-"], ["WW961", "A60|S250-"], ["WW959", "A60|S250-"], ["WW957", "A60-|S250-"], ["WW955", "A60-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW964", "A70-|S250-"], ["WW956", "A60|S250-"], ["WW954", "A60-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"BALAD":`

File: assets/airports/loww.json [hunk 6]
```diff
@@ -590,244 +572,94 @@
...
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW966", "A60|S250-"], ["WW956", "A60-|S250-"], ["WW954", "A60-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"PESAT":`

File: assets/airports/loww.json [hunk 7]
```diff
@@ -590,244 +572,94 @@
...
+          "body": [["WW953", "A60-|S250-"], ["WW952", "A30|S250-"], ["WW951", "A30-|S250-"], ["WW950", "A30-|S250-"], ["RW29", "A50-|S250-"]]
+        },
+        "16": {
+          "icao": "16",
+          "name": "runway 16",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A60+|S250-"], ["WW681", "A100-|S250-"], ["WW679", "A60|S250-"], ["WW677", "A60-|S250-"], ["WW675", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW690", "A100|S250-"], ["WW688", "A80-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW686", "A60|S250-"], ["WW684", "A60|S250-"], ["WW682", "A60-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50-|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW684", "A80|S250-"], ["WW682", "A60+|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"PESAT":`

File: assets/airports/loww.json [hunk 8]
```diff
@@ -590,244 +572,94 @@
...
+          "body": [["WW673", "A50-|S250-"], ["WW672", "A50-|S250-"], ["WW671", "A50-|S250-"], ["WW670", "A50-|S250-"], ["RW16", "A50-|S250-"]]
+        },
+        "34": {
+          "icao": "34",
+          "name": "runway 34",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A70+|S250-"], ["WW987", "A70+|S250-"], ["WW985", "A70|S250-"], ["WW983", "A70|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW987", "A100|S250-"], ["WW985", "A80|S250-"], ["WW983", "A80|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW988", "A80|S250-"], ["WW982", "A50-|S250-"], ["WW980", "A60-|S250-"], ["WW978", "A50-|S250-"], ["WW976", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW991", "A70|S250-"], ["WW989", "A60+|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"PESAT":`

File: assets/airports/loww.json [hunk 9]
```diff
@@ -590,244 +572,94 @@
...
+            },
+          "body": [["WW673", "A50-|S250-"], ["WW672", "A50-|S250-"], ["WW671", "A50-|S250-"], ["WW670", "A50-|S250-"], ["RW16", "A50-|S250-"]]
+        },
+        "34": {
+          "icao": "34",
+          "name": "runway 34",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A70+|S250-"], ["WW987", "A70+|S250-"], ["WW985", "A70|S250-"], ["WW983", "A70|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW987", "A100|S250-"], ["WW985", "A80|S250-"], ["WW983", "A80|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW988", "A80|S250-"], ["WW982", "A50-|S250-"], ["WW980", "A60-|S250-"], ["WW978", "A50-|S250-"], ["WW976", "A50-|S250-"], ["WW974", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"BALAD":`

File: assets/airports/loww.json [hunk 10]
```diff
@@ -590,244 +572,94 @@
...
+            },
+          "body": [["WW953", "A60-|S250-"], ["WW952", "A30|S250-"], ["WW951", "A30-|S250-"], ["WW950", "A30-|S250-"], ["RW29", "A50-|S250-"]]
+        },
+        "16": {
+          "icao": "16",
+          "name": "runway 16",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A60+|S250-"], ["WW681", "A100-|S250-"], ["WW679", "A60|S250-"], ["WW677", "A60-|S250-"], ["WW675", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW690", "A100|S250-"], ["WW688", "A80-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW686", "A60|S250-"], ["WW684", "A60|S250-"], ["WW682", "A60-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50-|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing colon, should be `"BALAD":`

File: assets/airports/loww.json [hunk 11]
```diff
@@ -590,244 +572,94 @@
...
+        },
+        "29": {
+          "icao": "29",
+          "name": "runway 29",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A60+|S250-"], ["WW963", "A70+|S250"], ["WW961", "A70|S250-"], ["WW959", "A60|S250-"], ["WW957", "A60-|S250-"], ["WW955", "A60-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW967", "A60+|S250-"], ["WW965", "A60+|S250-"], ["WW961", "A60|S250-"], ["WW959", "A60|S250-"], ["WW957", "A60-|S250-"], ["WW955", "A60-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW964", "A70-|S250-"], ["WW956", "A60|S250-"], ["WW954", "A60-|S250-"]],
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW966", "A60|S250-"], ["WW956", "A60-|S250-"], ["WW954", "A60-|S250-"]],
+              "MASUR": ["_MASUR272009", "MASUR", "A170-|S330"], ["TEMTA", "A60-|S250"], ["NERDU", "A60+|S250-"], ["WW963", "A70+|S250"], ["WW961", "A70|S250-"], ["WW959", "A60|S250-"], ["WW957", "A60-|S250-"], ["WW955", "A60-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing bracket, should be `["MASUR, "A170-|S330"]`

File: assets/airports/loww.json [hunk 12]
```diff
@@ -590,244 +572,94 @@
...
+        },
+        "16": {
+          "icao": "16",
+          "name": "runway 16",
+          "entryPoints": {
+              "NERDU": [["NERDU", "A60+|S250-"], ["WW681", "A100-|S250-"], ["WW679", "A60|S250-"], ["WW677", "A60-|S250-"], ["WW675", "A50-|S250-"]],
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW690", "A100|S250-"], ["WW688", "A80-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW686", "A60|S250-"], ["WW684", "A60|S250-"], ["WW682", "A60-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50-|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW684", "A80|S250-"], ["WW682", "A60+|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]],
+              "MASUR": ["_MASUR272009", "MASUR", "A170-|S330"], ["TEMTA", "A60-|S250"], ["NERDU", "A60+|S250-"], ["WW681", "A100-|S250-"], ["WW679", "A60|S250-"], ["WW677", "A60-|S250-"], ["WW675", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing bracket, should be `["MASUR, "A170-|S330"]`

File: assets/airports/loww.json [hunk 13]
```diff
@@ -590,244 +572,94 @@
...
+              "LANUX": ["_LANUX140000", ["LANUX", "A150-|S330"], ["NERDU", "A60+|S250-"], ["WW681", "A100-|S250-"], ["WW679", "A60|S250-"], ["WW677", "A60-|S250-"], ["WW675", "A50-|S250-"]],
+              "GAMLI": ["LIMRA", ["GAMLI", "A60+|S330+"], ["BARUG", "A170-|S330"], ["WW450", "A90-|S250-"], ["BALAD", "A100|S250-"], ["WW686", "A60|S250-"], ["WW684", "A60|S250-"], ["WW682", "A60-|S250-"], ["WW680", "A60|S250-"], ["WW678", "A50-|S250-"], ["WW676", "A50-|S250-"], ["WW674", "A50-|S250-"]
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing bracket at the end of the line

Comment (indianbhaji) [inline assets/airports/loww.json]: (and comma)

**Comment by jakobeng1303** [comment]
Thank you @indianbhaji !

File: assets/airports/loww.json [hunk 14]
```diff
@@ -590,244 +572,94 @@
...
+              "MABOD": [["MABOD", "A60+|S250-"], ["WW987", "A100|S250-"], ["WW985", "A80|S250-"], ["WW983", "A80|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "BALAD"  [["BALAD", "A100|S250-"], ["WW988", "A80|S250-"], ["WW982", "A50-|S250-"], ["WW980", "A60-|S250-"], ["WW978", "A50-|S250-"], ["WW976", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "PESAT"  [["PESAT", "A60+|S250-"], ["WW991", "A70|S250-"], ["WW989", "A60+|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "MASUR": ["_MASUR272009", "MASUR", "A170-|S330"], ["TEMTA", "A60-|S250"], ["NERDU", "A70+|S250-"], ["WW987", "A70+|S250-"], ["WW985", "A70|S250-"], ["WW983", "A70|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
```
Comment (indianbhaji) [inline assets/airports/loww.json]: Missing bracket, should be `["MASUR, "A170-|S330"]`

**Comment by jakobeng1303** [comment]
I'm sorry for all the mistakes made

**Comment by n8rzz** [comment]
@jakobeng1303 the heroku app won't build until the longing issues are resolved in your airport file. A syntax error is causing the build to fail.

**Comment by jakobeng1303** [comment]
good to know

File: assets/airports/loww.json [hunk 15]
```diff
@@ -590,244 +572,94 @@
...
+              "NATEX": ["_NATEX080019", ["NATEX", "A140-|S330-"], ["PESAT", "A60+|S250-"], ["WW991", "A70|S250-"], ["WW989", "A60+|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "MIKOV": ["_MIKOV345020", ["MIKOV", "A170-|S330"], ["MABOD", "A60+|S250-"], ["WW987", "A100|S250-"], ["WW985", "A80|S250-"], ["WW983", "A80|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]],
+              "REKLU": ["_REKLU262000", ["REKLU", "A180|S250-"], ["MABOD", "A60+|S250-"], ["WW987", "A100|S250-"], ["WW985", "A80|S250-"], ["WW983", "A80|S250-"], ["WW981", "A50-|S250-"], ["WW979", "A50-|S250-"], ["WW977", "A50-|S250-"], ["WW975", "A50-|S250-"], ["WW974", "A50-|S250-"]]
+            },
+          "body": [["WW673", "A50-|S250-"], ["WW672", "A50-|S250-"], ["WW671", "A50-|S250-"], ["WW670", "A50-|S250-"], ["RW16", "A50-|S250-"]]
```
Comment (indianbhaji) [inline assets/airports/loww.json]: `rwy` section required, this is why this method doesn't work really

**Comment by jakobeng1303** [comment]
I have been testing it now for around 40 Minutes and it's great! I had some planes that dint wanted to leave the hold for some reason and had to kill them but everything else worked just fine.
@indianbhaji couch I just delete the `body` section and put the fixes in the according `entry point` list or should I replace body with the according runway and put the these final fixes there? So far I think this a very good method for european stars and transitions because you don't have to type that much and you don't have to know all the STARS

**Comment by indianbhaji** [comment]
Why are we putting aircraft directly onto a transition though? This definitely would not work well with Frankfurt as it's completely unrealistic for Frankfurt...?

**Comment by jakobeng1303** [comment]
What do you mean with putting acft directly onto a transition and how is that not realistic for Frankfurt?

**Comment by indianbhaji** [comment]
@jakobeng1303 Putting them on a route that takes them directly to an ILS without clearance which is the case here. At Frankfurt, you're either manually cleared put on a transition or vectored towards the runway.

**Comment by jakobeng1303** [comment]
in Frankfurt you would get clearance from Center not form Approach though so actually it's more unrealistic in Vienna (you would get clearance for the transition from approach)

**Comment by indianbhaji** [comment]
At Frankfurt, clearance for the STAR is given by centre - that's fine, but the transition waypoints are under approach's airspace so transition clearance is given by approach - it's the case with pretty much all German airports

**Comment by indianbhaji** [comment]
Transitions are coming under a separate issue so it might be worth just waiting until that feature is added

**Comment by jakobeng1303** [comment]
https://www.youtube.com/watch?v=97UarwKILzI 
I might be wrong but it's actually center by the look of it that clears them on the transition! No, I don't want to wait for this feature because I don't see why! If the "pre-clearance" is the issue, that it's just a matter of bringing back those STARS into the clearance limit fixes and clearing acft form there but tbh I'm not supporting this unless routes have automatic holds

**Comment by indianbhaji** [comment]
Put an @ before the fix name to get a hold (I believe)

**Comment by jakobeng1303** [comment]
@indianbhaji does that officially work now?

**Comment by indianbhaji** [comment]
@jakobeng1303 none at the moment, https://github.com/openscope/openscope/pull/161 < see Erik's comment on there

**Comment by jakobeng1303** [comment]
@indianbhaji very interesting indeed but changing the system back doesn't take an awful lot of work and so far this is the best way of doing it. What I will do however is leaving the body section empty and inserting the body fixes into the entry section. This way a plane is guaranteed to be able to follow the whole route

**Comment by jakobeng1303** [comment]
so what's gonna happen now?

**Comment by erikquinn** [comment]
The PR needs to be reviewed, and we will go back and forth with feedback until the airport is fully in compliance with the [specs](https://github.com/openscope/openscope/blob/develop/documentation/airport-file-standards.md).

You can track the status of any airport's compliance with any of the specs at [bit.ly/openscopeairports](http://bit.ly/openscopeairports).

This PR is actually next in line, so I'll be starting the review shortly. May take a few days, may take a few hours; depends on how much time I have and the state of the airport. So stay tuned! :smiley:

**Comment by erikquinn** [comment]
Closing in favor of new branch (same content and history) with the proper name of `feature/474`.

New pull request is here: #518.

**Comment by jakobeng1303** [comment]
ok thanks for the heads up!

**Pull request closed without merging**